src/communication_managers/iothub_communication_manager/inc/aduc/iothub_communication_manager.h

```diff
@@ -13,6 +13,40 @@
 
 EXTERN_C_BEGIN
 
+/**
+ * @brief Scans the connection string and returns the connection type related to the string
+ * @details The connection string must use the valid, correct format for the DeviceId and/or the ModuleId
+ * e.g.
+ * "DeviceId=some-device-id;ModuleId=some-module-id;"
+ * If the connection string contains the DeviceId it is an ADUC_ConnType_Device
+ * If the connection string contains the DeviceId AND the ModuleId it is an ADUC_ConnType_Module
+ * @param connectionString the connection string to scan
+ * @returns the connection type for @p connectionString
+ */
+ADUC_ConnType GetConnTypeFromConnectionString(const char* connectionString);
+
+/**
+ * @brief Get the Connection Info from connection string, if a connection string is provided in configuration file
+ *
+ * @return true if connection info can be obtained
+ */
+bool GetConnectionInfoFromConnectionString(ADUC_ConnectionInfo* info, const char* connectionString);
```

Comment:
Are these used outside of iothub_communication_manager.c?  If not, there is no need to expose them externally (it can prevent linker from removing unused code)